nitransforms/io/lta.py

```diff
@@ -39,10 +39,10 @@ def as_affine(self):
         """Return the internal affine of this regular grid."""
         affine = np.eye(4)
         sa = self.structarr
-        A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
-        b = sa["cras"] - A.dot(sa["volume"]) / 2
-        affine[:3, :3] = A
-        affine[:3, [3]] = b
+        affine[:3, :3] = (
+            np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"].flatten()
+        )
+        affine[:3, [3]] = sa["cras"] - affine[:3, :3] @ sa["volume"] / 2
         return affine
```

Comment:
I feel like this loses clarity. Right now we're decomposing `Mx = Ax + b`, and showing how A and b interact.

Also, if I'm understanding you correctly, the problem is that we're applying `voxelsize` row-wise when it should be column-wise, which suggests the correct fix is:

```Diff
-        A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
+        A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) @ sa["voxelsize"]
```

The reason that the header fields are shaped as they are is to make the computations straightforward linear algebra. If it's not useful, we can just flatten them into shape `(3,)` instead of `(3, 1)`.

Finally, however we do this, could we add a test with a geometry that shows the old formulation was wrong and the new correct? I get the regression tests are checking this, but something readable, with, e.g. voxel sizes like `(2, 3, 4)` and cosines that are ones/zeros would be nice.

Reply:
```Python
A = np.hstack((sa["xras"], sa["yras"], sa["zras"])) * sa["voxelsize"]
```
If the dimensions of `sa["voxelsize"]` are (3, 1) then this is row-wise product. Conversely, if the dimension is (3, ) then this is col-wise product. We want column wise product and before this patch we were doing row wise product.

Dot product doesn't apply here, as it rotates the voxelsize vector through the affine.